File: cdk/lambda/client/function.js
const MAX_SLEEP = 3*1000;
const ERROR_RATE = 10;
const THROW_ERRORS = process.env.THROW_ERRORS || false;

console.log(`Throw error is ${(THROW_ERRORS) ? 'enabled':'disabled'}`)

exports.handler = async (event) => {
    if(THROW_ERRORS) {
        throwError();
    }

    const time = randomSleep();
    console.log(`Sleeping ${time} ms`);

    await sleep(time);

    const response = {
        statusCode: 200,
        body: JSON.stringify('Hello from Lambda!'),
    };

    return response;
};

// throw randomized error based on ERROR_RATE constant
function throwError() {
    if (Math.floor(Math.random() * 100) < ERROR_RATE) {
        throw new Error('Oh no! We got an error :(');
    }
}

// randomize a number between 0 and MAX_SLEEP constant
function randomSleep() {
    return Math.floor(Math.random() * MAX_SLEEP);
}

// sleep the code
async function sleep(ms) {
    return new Promise((resolve) => {
        setTimeout(resolve, ms);
    });
}

File: cdk/lambda/client-python/function.py
import os
import json
import random
import time

MAX_SLEEP = 3 * 1000
ERROR_RATE = 10
THROW_ERRORS = os.environ.get('THROW_ERRORS') == 'true'

print(f'Throw error is {"enabled" if THROW_ERRORS else "disabled"}')

def handler(event, context):
    if THROW_ERRORS:
        throw_error()

    time_to_sleep = random_sleep()
    print(f'Sleeping {time_to_sleep} ms')

    time.sleep(time_to_sleep / 1000)  # Sleep time is in seconds

    response = {
        'statusCode': 200,
        'body': json.dumps('Hello from Lambda!')
    }

    return response

def throw_error():
    if random.randint(1, 100) <= ERROR_RATE:
        raise Exception('Oh no! We got an error :(')

def random_sleep():
    return random.randint(0, MAX_SLEEP)